### ISSUE TITLE
Request for feature in @JSProperty

### ISSUE THREAD

**Issue by ulrd** [konsoletyper/teavm#496]
Hello,

When an instance is created in java the class members are always initialized. There is no equivalent to 'undefined' in java world. It is therefore surprising for a java code to fail when executing such a piece of code:
```
if (foo.getBar() == null) {
    // When bar is null
}
else {
    // When bar is not null
}
```

Yet this code stops with an error when running in a browser if the foo instance is a JSObject with undefined/uninitialized bar attribute. I would have expected the transpilation to take care of this details, like GWT does. My current workaround is to never use @JSProperty for getters but a @JSBody with a check. This is cumbersome.

I'd like to propose an additional String parameter defaultValue to the @JSProperty annotation. To keep backward compatibility its default value would be the empty string. The meaning of this parameter is the default value returned when the attribute is undefined.

If @JSProperty.defaultValue() is "" the code generated is equivalent to 
`@JSBody(script = "return this.propertyName;")` 
When it is not empty it is
`@JSBody(script = "return this.propertyName===undefined ? <the_default_value> : this.propertyName;")`

A @JSSafeProperty is welcome as a shortcut to @JSProperty(defaultValue="null")

Thanks for considering

**Comment by konsoletyper** [comment]
`null` and `undefined` have different meaning in JavaScript and it's not always `null` in Java. Sometimes you need to distinguish between facts: property is present but its value is `null` and property isn't in object (also there may be situation when property exists but its value is `undefined`). There might be different scenarios of handling `undefined` values and I don't think we should bind TeaVM interop layer to one distinct. You can check if value is `undefined` by calling `JSObjects.isUndefined`. Also, you can write method like following:

```java
<T extends JSObject> T guardUndefined(T value, T default) {
    return JSObjects.isUndefined(value) ? default : value;
}
```

**Comment by ulrd** [comment]
I fully agree on the fact that `null` and `undefined` have different meanings in JavaScript. But there is no meaning for an undefined class member in java. I understand TeaVM as a java to JS transpiler (=java semantics come first). Getting an undefined value crashes the script with an error. This is not desirable IMHO.

The distinction between `null` and not set is indeed interesting in a `Map.containsKey()` scenario. I wouldn't lose this for sure. Yet this is more the exception than the rule.

I also agree that TeaVM shall not prevent the user from checking the unset state to keep close to javascript for low-level interaction. This is why I suggested an **optional** `defaultValue`; _backward compatible with the current situation_.

Also consider people like me comming from GWT (I'm converting a GWT project). GWT overlays are much more flexible. TeaVM gave me a hard time to convert ~60 JavaScriptObject to JSObect. I indeed had to use your proposed workaround (but using a JS global function instead). I can assure you the result is ugly when used hundreds of times.
